br/pkg/lightning/backend/kv/session.go

```diff
@@ -287,11 +288,14 @@ func NewSession(options *encode.SessionOptions, logger log.Logger) *Session {
 	vars.StmtCtx.InInsertStmt = true
 	vars.StmtCtx.BatchCheck = true
 	vars.StmtCtx.BadNullAsWarning = !sqlMode.HasStrictMode()
-	vars.StmtCtx.TruncateAsWarning = !sqlMode.HasStrictMode()
 	vars.StmtCtx.OverflowAsWarning = !sqlMode.HasStrictMode()
 	vars.StmtCtx.AllowInvalidDate = sqlMode.HasAllowInvalidDatesMode()
 	vars.StmtCtx.IgnoreZeroInDate = !sqlMode.HasStrictMode() || sqlMode.HasAllowInvalidDatesMode()
 	vars.SQLMode = sqlMode
+
+	var typeConvFlags types.Flags
+	typeConvFlags = typeConvFlags.WithTruncateAsWarning(!sqlMode.HasStrictMode())
+	vars.StmtCtx.TypeConvContext = types.NewContext(typeConvFlags, vars.Location(), vars.StmtCtx.AppendWarning)
```

Comment:
or function name `SetTypeFlags` is better

Reply:
Creating a new `TypeConvContext` is fine (at least here), because the session vars and stmtctx are also newly created.

Maybe we can have a `InitTypeConvContext` to derive `TypeConvContext` from other session vars and stmts (or other necessary states) in the future to help other developers to initialize the `TypeConvContext` correctly, and re-use this function across many different `newSession`, `NewSession` and `ResetContextOfStmt`...

But for only `SetTypeFlags` or `ResetTypeContext`, I don't think it's much better than a single line `NewContext` 🤔 .

Reply:
Changed to use `SetTypeFlags(sc.TypeFlags().WithXXX)